### ISSUE TITLE
Scala.js linkingErrors on Uri.addParams(map)

### ISSUE THREAD

**Issue by exoego** [lemonlabsuk/scala-uri#160]
## Describe the bug 

* scala-uri version: 2.2.2
* ScalaJS version: 1.0.1 & 1.1.0
* Scala version: 2.13.2

`Uri.addParams(map)` worked on Scala.js 0.6.x, but on Scala.js 1.0.0/1.1.0, it start failiing when js-linkning (`fastOptJS`/`fullOptJS`) time.

Not sure, but maybe related to #152 

## To Reproduce

```scala
val params = Map.empty[String,String]
val withParams = Url.parse("/foo/bar").addParams(params)
```

<details>
<summary>stack trace</summary>

```
[error] Referring to non-existent class io.lemonlabs.uri.typesafe.TraversableParams$Ops
[error]   called from io.lemonlabs.uri.typesafe.TraversableParams$AllOps
[error]   called from io.lemonlabs.uri.typesafe.TraversableParams$ops$.toAllTraversableParamsOps(java.lang.Object,io.lemonlabs.uri.typesafe.TraversableParams)io.lemonlabs.uri.typesafe.TraversableParams$AllOps
[error]   called from io.lemonlabs.uri.QueryString.addParams(java.lang.Object,io.lemonlabs.uri.typesafe.TraversableParams)io.lemonlabs.uri.QueryString
[error]   called from io.lemonlabs.uri.Url.addParams(java.lang.Object,io.lemonlabs.uri.typesafe.TraversableParams)io.lemonlabs.uri.Url
[error]   called from Foo$.main([java.lang.String)void
[error]   called from static Foo.main([java.lang.String)void
[error]   called from core module module initializers
[error] involving instantiated classes:
[error]   io.lemonlabs.uri.typesafe.TraversableParams$ops$
[error]   io.lemonlabs.uri.QueryString
[error]   Foo$
[error] Referring to non-existent method io.lemonlabs.uri.typesafe.TraversableParams$AllOps.toSeq()scala.collection.immutable.Seq
[error]   called from io.lemonlabs.uri.QueryString.addParams(java.lang.Object,io.lemonlabs.uri.typesafe.TraversableParams)io.lemonlabs.uri.QueryString
[error]   called from io.lemonlabs.uri.Url.addParams(java.lang.Object,io.lemonlabs.uri.typesafe.TraversableParams)io.lemonlabs.uri.Url
[error]   called from Foo$.main([java.lang.String)void
[error]   called from static Foo.main([java.lang.String)void
[error]   called from core module module initializers
[error] involving instantiated classes:
[error]   io.lemonlabs.uri.QueryString
[error]   Foo$
[error] Referring to non-existent class io.lemonlabs.uri.typesafe.TraversableParams$Ops
[error]   called from constructor io.lemonlabs.uri.typesafe.TraversableParams$ops$$anon$11.<init>(java.lang.Object,io.lemonlabs.uri.typesafe.TraversableParams)void
[error]   called from io.lemonlabs.uri.typesafe.TraversableParams$ops$.toAllTraversableParamsOps(java.lang.Object,io.lemonlabs.uri.typesafe.TraversableParams)io.lemonlabs.uri.typesafe.TraversableParams$AllOps
[error]   called from io.lemonlabs.uri.QueryString.addParams(java.lang.Object,io.lemonlabs.uri.typesafe.TraversableParams)io.lemonlabs.uri.QueryString
[error]   called from io.lemonlabs.uri.Url.addParams(java.lang.Object,io.lemonlabs.uri.typesafe.TraversableParams)io.lemonlabs.uri.Url
[error]   called from Foo$.main([java.lang.String)void
[error]   called from static Foo.main([java.lang.String)void
[error]   called from core module module initializers
[error] involving instantiated classes:
[error]   io.lemonlabs.uri.typesafe.TraversableParams$ops$
[error]   io.lemonlabs.uri.QueryString
[error]   Foo$
[error] Referring to non-existent method io.lemonlabs.uri.typesafe.TraversableParams$Ops.$init$()void
[error]   called from constructor io.lemonlabs.uri.typesafe.TraversableParams$ops$$anon$11.<init>(java.lang.Object,io.lemonlabs.uri.typesafe.TraversableParams)void
[error]   called from io.lemonlabs.uri.typesafe.TraversableParams$ops$.toAllTraversableParamsOps(java.lang.Object,io.lemonlabs.uri.typesafe.TraversableParams)io.lemonlabs.uri.typesafe.TraversableParams$AllOps
[error]   called from io.lemonlabs.uri.QueryString.addParams(java.lang.Object,io.lemonlabs.uri.typesafe.TraversableParams)io.lemonlabs.uri.QueryString
[error]   called from io.lemonlabs.uri.Url.addParams(java.lang.Object,io.lemonlabs.uri.typesafe.TraversableParams)io.lemonlabs.uri.Url
[error]   called from Foo$.main([java.lang.String)void
[error]   called from static Foo.main([java.lang.String)void
[error]   called from core module module initializers
[error] involving instantiated classes:
[error]   io.lemonlabs.uri.typesafe.TraversableParams$ops$
[error]   io.lemonlabs.uri.QueryString
[error]   Foo$
[error] Referring to non-existent class io.lemonlabs.uri.typesafe.TraversableParams$Ops
[error]   called from io.lemonlabs.uri.typesafe.TraversableParams$ops$$anon$11.toSeq()scala.collection.immutable.Seq
[error]   called from io.lemonlabs.uri.QueryString.addParams(java.lang.Object,io.lemonlabs.uri.typesafe.TraversableParams)io.lemonlabs.uri.QueryString
[error]   called from io.lemonlabs.uri.Url.addParams(java.lang.Object,io.lemonlabs.uri.typesafe.TraversableParams)io.lemonlabs.uri.Url
[error]   called from Foo$.main([java.lang.String)void
[error]   called from static Foo.main([java.lang.String)void
[error]   called from core module module initializers
[error] involving instantiated classes:
[error]   io.lemonlabs.uri.typesafe.TraversableParams$ops$$anon$11
[error]   io.lemonlabs.uri.QueryString
[error]   Foo$
[error] Referring to non-existent method io.lemonlabs.uri.typesafe.TraversableParams$Ops.toSeq()scala.collection.immutable.Seq
[error]   called from io.lemonlabs.uri.typesafe.TraversableParams$ops$$anon$11.toSeq()scala.collection.immutable.Seq
[error]   called from io.lemonlabs.uri.QueryString.addParams(java.lang.Object,io.lemonlabs.uri.typesafe.TraversableParams)io.lemonlabs.uri.QueryString
[error]   called from io.lemonlabs.uri.Url.addParams(java.lang.Object,io.lemonlabs.uri.typesafe.TraversableParams)io.lemonlabs.uri.Url
[error]   called from Foo$.main([java.lang.String)void
[error]   called from static Foo.main([java.lang.String)void
[error]   called from core module module initializers
[error] involving instantiated classes:
[error]   io.lemonlabs.uri.typesafe.TraversableParams$ops$$anon$11
[error]   io.lemonlabs.uri.QueryString
[error]   Foo$
[error] There were linking errors
[error] (Compile / fastOptJS) There were linking errors
[error] Total time: 1 s, completed 2020/06/02 11:44:59
```
</details>

### Easy way 

I created a repo for minimum reproduction code with Scala.js 1.1.0 setup.

```bash
git clone https://github.com/exoego/scalajs1-errors.git --branch scala-uri-2.2.2
cd scalajs1-errors
sbt fastOptJS
```

**Comment by theon** [comment]
Thanks for raising. Yes, looks like the same issue as https://github.com/lemonlabsuk/scala-uri/issues/152 to me. I managed to work around the issue by removing simulacrum, but that's not really a viable fix

**Comment by theon** [comment]
This has been released as version `2.2.3` and should be in maven central shortly
